compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1592,6 +1610,11 @@ impl<'tcx> Place<'tcx> {
         self.projection.iter().any(|elem| elem.is_indirect())
     }
 
+    /// Returns `true` if this `Place` contains a `Subtype` projection.
+    pub fn is_subtype(&self) -> bool {
```

Comment:
i would expect these `is_subtype` methods to not be needed 🤔

Reply:
this is still needed, when we are creating move paths.